unity/src/test/scala/io/delta/unity/UCCatalogManagedTestUtils.scala

```diff
@@ -99,4 +103,64 @@ trait UCCatalogManagedTestUtils extends TestUtils with ActionUtils with WriteUti
     val simpleRow = DataBuilderUtils.row(schema, elem)
     singletonCloseableIterator(simpleRow)
   }
+
+  /** Version TS for the test table used in [[withUCClientAndTestTable]] */
+  val v0Ts = 1749830855993L // published commit
+  val v1Ts = 1749830871085L // ratified staged commit
+  val v2Ts = 1749830881799L // ratified staged commit
+
+  /**
+   * @param textFx test function to run that takes input (ucClient, tablePath, maxRatifiedVersion)
+   */
+  def withUCClientAndTestTable(
+      textFx: (InMemoryUCClientWithMetrics, String, Long) => Unit): Unit = {
+    val maxRatifiedVersion = 2L
+    val tablePath = getTestResourceFilePath("catalog-owned-preview")
+    val ucClient = new InMemoryUCClientWithMetrics("ucMetastoreId")
+    val fs = FileSystem.get(new Configuration())
+    val catalogCommits = Seq(
+      // scalastyle:off line.size.limit
+      getTestResourceFilePath("catalog-owned-preview/_delta_log/_staged_commits/00000000000000000001.4cb9708e-b478-44de-b203-53f9ba9b2876.json"),
+      getTestResourceFilePath("catalog-owned-preview/_delta_log/_staged_commits/00000000000000000002.5b9bba4a-0085-430d-a65e-b0d38c1afbe9.json"))
+      // scalastyle:on line.size.limit
+      .map { path => fs.getFileStatus(new Path(path)) }
+      .map { fileStatus =>
+        new Commit(
+          FileNames.deltaVersion(fileStatus.getPath.toString),
+          fileStatus,
+          fileStatus.getModificationTime)
+      }
+    val tableData = new TableData(maxRatifiedVersion, ArrayBuffer(catalogCommits: _*))
+    ucClient.createTableIfNotExistsOrThrow("ucTableId", tableData)
+    textFx(ucClient, tablePath, maxRatifiedVersion)
+  }
+
+  /** Wrapper class around InMemoryUCClient that tracks number of getCommit calls made */
+  class InMemoryUCClientWithMetrics(ucMetastoreId: String) extends InMemoryUCClient(ucMetastoreId) {
```

Comment:
I like to keep classes either at the bottom or top -- not in the middle -- that gets messy as now inner classes are mixed in with methods